kbcstorage/configurations.py

```diff
@@ -0,0 +1,114 @@
+            configuration_id (str): The id of the configuration.
+
+        Returns:
+            response_body: The parsed json from the HTTP response.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        if not isinstance(component_id, str) or component_id == '':
+            raise ValueError("Invalid component_id '{}'.".format(component_id))
+        if not isinstance(configuration_id, str) or configuration_id == '':
+            raise ValueError("Invalid component_id '{}'.".format(configuration_id))
+        url = '{}/{}/configs/{}'.format(self.base_url, component_id, configuration_id)
+        return self._get(url)
+
+    def delete(self, component_id, configuration_id):
+        """
+        Deletes the configuration.
+
+        Args:
+            component_id (str): The id of the component.
+            configuration_id (str): The id of the configuration.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        if not isinstance(component_id, str) or component_id == '':
+            raise ValueError("Invalid component_id '{}'.".format(component_id))
+        if not isinstance(configuration_id, str) or configuration_id == '':
+            raise ValueError("Invalid component_id '{}'.".format(configuration_id))
+        url = '{}/{}/configs/{}'.format(self.base_url, component_id, configuration_id)
+        self._delete(url)
+
+    def list(self, component_id):
+        """
+        Lists configurations of the given component.
+
+        Args:
+            component_id (str): The id of the component.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        if not isinstance(component_id, str) or component_id == '':
+            raise ValueError("Invalid component_id '{}'.".format(component_id))
+        url = '{}/{}/configs'.format(self.base_url, component_id)
+        return self._get(url)
+
+    def create(self, component_id, name, description='', configuration=None, state=None, change_description='',
+               is_disabled=False, configuration_id=None):
+        """
+        Create a new configuration.
+
+        Args:
+            component_id (str): ID of the component to create configuration for.
+            name (str): Name of the configuration visible to end-user.
+            description (str): Optional configuration description
+            configuration (dict): Actual configuration parameters
+            state (dict): Optional state parameters
+            changeDescription (str): Optional change description
+            is_disabled (bool): Optional flag to disable the configuration, default False
+            configuration_id (str): Optional configuration ID, if not specified, new ID is generated
+        Returns:
+            response_body: The parsed json from the HTTP response.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
```

Comment:
Here we should also validate `component_id` right?

Reply:
right, added